src/esw/brushless.py

```diff
@@ -0,0 +1,219 @@
+        _ratio_motor_to_wheel = rospy.get_param("wheel/gear_ratio")
+        self._wheels_m_s_to_motor_rad_ratio = (1 / wheel_radius) * _ratio_motor_to_wheel
+
+        _max_speed_m_s = rospy.get_param("rover/max_speed")
+        assert _max_speed_m_s > 0, "rover/max_speed config must be greater than 0"
+
+        max_motor_rad_s = _max_speed_m_s * self._wheels_m_s_to_motor_rad_ratio
+        measured_max_motor_rad_s = 1 * 2 * math.pi  # Should not be changed. Derived from testing.
+        self._max_motor_speed_rad_s = min(measured_max_motor_rad_s, max_motor_rad_s)
+
+        rospy.Subscriber("cmd_vel", Twist, self._process_twist_message)
+
+    def _process_twist_message(self, ros_msg: Twist):
+        """Converts the twist message into rad/s to turn per wheel.
+        Then tells the wheels to move at that speed.
+        :param ros_msg: A Twist message of the rover to turn."""
+        forward = ros_msg.linear.x
+        turn = ros_msg.angular.z
+
+        turn_difference_inner = turn * self._wheel_distance_inner
+        turn_difference_outer = turn * self._wheel_distance_outer
+
+        left_rad_inner = (forward - turn_difference_inner) * self._wheels_m_s_to_motor_rad_ratio
+        right_rad_inner = (forward + turn_difference_inner) * self._wheels_m_s_to_motor_rad_ratio
+        left_rad_outer = (forward - turn_difference_outer) * self._wheels_m_s_to_motor_rad_ratio
+        right_rad_outer = (forward + turn_difference_outer) * self._wheels_m_s_to_motor_rad_ratio
+
+        # Ignore inner since outer > inner always
+        larger_abs_rad_s = max(abs(left_rad_outer), abs(right_rad_outer))
+        if larger_abs_rad_s > self._max_motor_speed_rad_s:
+            change_ratio = self._max_motor_speed_rad_s / larger_abs_rad_s
+            left_rad_inner *= change_ratio
+            right_rad_inner *= change_ratio
+            left_rad_outer *= change_ratio
+            right_rad_outer *= change_ratio
+
+        for name, bridge in self.drive_bridge_by_name.items():
+            bridge.start_time = t.process_time()
+
+            if name == "front_left" or name == "back_left":
+                commanded_velocity = left_rad_outer
+            elif name == "middle_left":
+                commanded_velocity = left_rad_inner
+            elif name == "front_right" or name == "back_right":
+                commanded_velocity = right_rad_outer
+            elif name == "middle_right":
+                commanded_velocity = right_rad_inner
+            else:
+                rospy.logerr(f"Invalid name {name}")
+                continue
+
+            if bridge.state == "Armed":
+                bridge.set_command(CommandData(position=math.nan, velocity=commanded_velocity, torque=0.5))
+
+    async def run(self):
+        previously_lost_communication = True
+        while not rospy.is_shutdown():
+            for name, bridge in self.drive_bridge_by_name.items():
+                if name != "middle_left" and name != "front_right":
```

Comment:
Might be nice to create a list of connected motors and just say `if name not in connected_motors` or something like that

Reply:
Oh ok, so I actually just forgot to remove this if statement lol